src/internal/throw.js

```diff
@@ -17,3 +18,29 @@ export const invalidContext = (it, actual) => typeError(
   `${it} was invoked outside the context of a Future. You might want to use`
   + ` a dispatcher instead\n  Called on: ${show(actual)}`
 );
+
+const invalidNamespace = (m, x) => (
+  `The Future was not created by ${namespace}. `
++ `Make sure you transform other Futures to ${namespace} Futures. `
++ `Got ${x ? `a Future from ${x}` : 'an unscoped Future'}.`
++ '\n  See: https://github.com/fluture-js/Fluture#casting-futures'
+);
+
+const invalidVersion = (m, x) => (
+  `The Future was created by ${x < version ? 'an older' : 'a newer'} version of ${namespace}. `
++ 'This means that one of the sources which creates Futures is outdated. '
++ 'Update this source, or transform its created Futures to be compatible.'
++ '\n  See: https://github.com/fluture-js/Fluture#casting-futures'
+);
+
+export const invalidFuture = (it, at, m, s = '') => {
```

Comment:
I know. This function has a pretty ugly signature: `(String, String | Number, Any, String | Nil) -> Void`. It's completely unoptimizable, but I don't care, because the only time this function is called is right before the process crashes.